packages/block-editor/src/components/inserter/block-patterns-tab/index.js

```diff
@@ -37,6 +38,10 @@ function BlockPatternsTab( {
 		[]
 	);
 
+	useEffect( () => {
+		onHasCategories( !! categories.length );
+	}, [ categories, onHasCategories ] );
```

Comment:
The `onHasCategories` prop will have different references on each render, which makes it unsafe to pass it to the effect.

I think the only reason we're not getting an infinite loop is that the state is a boolean value.

Reply:
Since we're close to the release, I took the liberty to fix this. I also changed the function name to just `setHasCategories` to better convey what it does. I hope it's okay for you, @talldan 🙇 !